blog_project/post/views.py

```diff
@@ -0,0 +1,299 @@
+
+        if comment is None:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        comment.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class ClapseList(APIView):
+
+    permission_class = (IsAuthorOrReadOnly, )
+
+    def get(self, request, pk, format=None):
+        clapse = Clapse.objects.filter(post__pk=pk)
+        clapse_created_by_ids = clapse.values('created_by')
+        print(clapse_created_by_ids)
+        created_by = user_models.objects.filter(
+            pk__in=clapse_created_by_ids)
+        serializer = user_serializers.UserSerializer(
+            created_by, many=True, context={'request': request}
+        )
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request, pk, format=None):
+        created_by = request.user
+
+        try:
+            find_post = Post.objects.get(pk=pk)
+        except:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            preexisting_clapse = Clapse.objects.get(
+                post=find_post,
+                created_by=created_by
+            )
+            return Response(status=status.HTTP_304_NOT_MODIFIED)
+        except:
+            clapse = Clapse.objects.create(
+                post=find_post,
+                created_by=created_by
+            )
+            clapse.save()
+            notification_views.create_notification(
+                created_by, find_post.created_by, 'like', find_post
+            )
+            return Response(status=status.HTTP_201_CREATED)
+
+
+class UnClapseList(APIView):
+
+    permission_class = (IsAuthorOrReadOnly, )
+
+    def delete(self, request, pk, format=None):
+        created_by = request.user
+
+        try:
+            find_post = Post.objects.get(pk=pk)
+        except:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            preexisting_clapse = Clapse.objects.get(
+                created_by=created_by,
+                post=find_post
+            )
+            preexisting_clapse.delete()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        except:
+            return Response(status=status.HTTP_304_NOT_MODIFIED)
+
+
+class TagPostList(APIView):
+
+    permission_class = (IsAuthorOrReadOnly, )
+    pagination_class = TagPagination
+
+    def get(self, request, tagname, format=None):
+
+        try:
+            tag = Tag.objects.get(name=tagname)
+        except Exception as e:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            post = Post.objects.filter(tags=tag)
+            serializer = PostListSerializer(
+                post, many=True, context={'request': request}
+            )
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            print(e)
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
특히 except Exception은 모든 exception을 catch하기 때문에 좋지 못한 패턴이에요.
except Model.DoesNotExist 등을 활용하는 편이 더 좋습니다.

```suggestion python
        except Post.DoesNotExist as e:
            return Response(status=status.HTTP_404_NOT_FOUND)
```